index.d.ts

```diff
@@ -1,54 +1,52 @@
-import * as mongodb from 'mongodb';
-import fastify, { FastifyPlugin } from 'fastify';
+import type { FastifyPlugin } from 'fastify';
+import type { Db, MongoClient, ObjectId } from 'mongodb';
 
-declare namespace fastifyMongodb {
-  interface FastifyMongoObject {
-    /**
-     * Mongo client instance
-     */
-    client: mongodb.MongoClient;
-    /**
-     * DB instance
-     */
-    db?: mongodb.Db;
-    /**
-     * Mongo ObjectId class
-     */
-    ObjectId: typeof mongodb.ObjectId;
-  }
+export interface FastifyMongoObject {
+  /**
+   * Mongo client instance
+   */
+  client: MongoClient;
+  /**
+   * DB instance
+   */
+  db?: Db;
+  /**
+   * Mongo ObjectId class
+   */
+  ObjectId: typeof ObjectId;
+}
 
-  interface FastifyMongoNestedObject {
-    [name: string]: FastifyMongoObject;
-  }
+export interface FastifyMongoNestedObject {
+  [name: string]: FastifyMongoObject;
+}
 
-  interface FastifyMongodbOptions {
-    /**
-     * Force to close the mongodb connection when app stopped
-     * @default false
-     */
-    forceClose?: boolean;
-    /**
-     * Database name to connect
-     */
-    database?: string;
-    name?: string;
-    /**
-     * Pre-configured instance of MongoClient
-     */
-    client?: mongodb.MongoClient;
-    /**
-     * Connection url
-     */
-    url?: string;
-  }
+export interface FastifyMongodbOptions {
```

Comment:
```suggestion
export interface FastifyMongodbOptions extends ConnectionOptions {
```

Reply:
as `mongodb@4` provide official plugin, we can use `ConnectionOptions` to allow users specify the connection options.